enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,609 @@
+- Early phases will introduce OLM v1 components that are in alpha stages.
+- No concrete migration plan from legacy to OLM v1.
+- Coordination around external team dependencies.
+- The BundleDeployment API has a large net of permissions.
+- The upstream rukpak project has a direct dependency on cert-manager.
+
+#### Mitigations
+
+The v1alpha1 CRDs won't be deployed by default, and those APIs will be hidden behind the "TechPreviewNoUpgrade" feature set that admins need to explicitly enable before cluster creation. Any new components introduced during these early phases will also specify the `release.openshift.io/feature-gate` annotation in their CVO manifests, and set that key's value to "TechPreviewNoUpgrade".
+This annotation will ensure that the CVO won't deploy any tech preview components.
+
+Instructing admins to configure upfront RBAC to limit the BundleDeployment API to the minimum set of users may help mitigate the plain provisioner's wide net of permissions. The process for constructing this upfront RBAC may require explicit documentation.
+
+We can mitigate the upstream rukpak project's direct dependency on the cert-manager operator by removing that dependency in favor of the downstream service-ca-operator component.
+
+### Drawbacks
+
+- Implementing this PO mechanism using legacy OLM would provide more immediate business value. That implementation is outlined below in the alternatives section.
+- Early phases will be delivered through tech preview guidelines. This leads to little customer feedback, given it precludes production clusters.
+- Early phases may introduce a new OLM 1.x control plane that has little-to-no interaction with the existing, legacy OLM 0.x control plane.
+- Early phases may introduce validating webhooks. These webhooks can impact the cluster's overall health when they're not available.
+- Early phases may introduce validating webhooks. By default, these webhooks will be deployed using an HA configuration, but may not have logic to accommodate SNO cluster environments.
+- Early phases may introduce a dynamic watching implementation for managing the underlying OLM operator resources. This has potential performance implications that need to be measured through benchmarking and/or addressed before any GA phase.
+- Future phase designs may require a large-scale rewrite of the OLM component.
+
+## Design Details
+
+### Open Questions [optional]
+
+- **Can future phase designs evaluate whether platform operators can be extended to cover things like the MAPI and CCM operators?**
+- Does having separate APIs for ClusterOperators and POs provide a degraded admin UX?
+- How to handle the maxOpenShiftVersion constraint given it's defined in the CSV resource?
+- Is it possible to have a dedicated PO index image?
+- How to prevent an admin installing an operator that was installed through the PO mechanism, and vice-versa.
+- How do admins determine the PO package name ahead of time?
```

Comment:
This feels like it's important to answer before we create an API based on that data

Reply:
We might have an answer to this already: it's a combination of documentation, and using existing clusters to determine the right package name during phase 0.

This is an obvious UX gap but it allows us to avoid any CVO/installer changes for this phase, which seems like a reasonable trade-off (imo) while OLM 1.x continues to be written. I'd imagine that phase 1 will start to interface with the installer component, and we'll be able to define a better UX to address these current gaps.

Reply:
Ok, perhaps we can capture that and move this to answered questions?

Reply:
Yup, that sounds reasonable to me.

Reply:
Added in https://github.com/openshift/enhancements/pull/1170/commits/31269f34570363358512ef7e14a4da757e779476.